### DISCUSSION TITLE
OCR Templater Script for Obsidian

### DISCUSSION THREAD

**Discussion by pmbauer** [SilentVoid13/Templater#217] [Templates Showcase]
[gist](https://gist.github.com/pmbauer/8fddd5e767ac87c99071819da52cb6c1)

# OCR Templater Script for Obsidian
Optical Character Recognition (OCR) [Templater][1] Script for [Obsidian][2].

## [Video Demonstration](https://twitter.com/pmbauer/status/1392574345407500291)

# Installation

### Pre-requisites
- [Obsidian][2]
- [Templater][1] plugin.

### Install the [Tesseract][3] OCR engine.
[Prebuilt Binaries for Linux and Windows][4] are available.  On Mac: `brew install tesseract`.

### Create Templater System Command for Tesseract

1. Open `Settings > Templater`
2. Turn on *Enable System Commands*
3. Add a *User Function* named `ocr`

```
/usr/local/bin/tesseract "$ocr_input" -
```

**NOTE** I reccomend increasing the Templater *Timeout* to 10s or more as the OCR may take a moment.

### Add OCR.md template
Add the [OCR.md](https://gist.githubusercontent.com/pmbauer/8fddd5e767ac87c99071819da52cb6c1/raw/9eb95f0b92e1f7a03acf20de252e4c0e8080cd29/OCR.md) template to your Templater templates folder.

```
---
creation date: <% tp.file.creation_date() %>
tags: [OCR]
---

<%*
const supportedFileTypes = ["jpeg", "jpg", "png"];
const images = this.app.vault.getFiles().filter((item) => supportedFileTypes.indexOf(item.extension) >= 0)
const target = await tp.system.suggester((item) => item.path, images, true);
const out = await tp.user.ocr({ocr_input: target.path});
%><%* tR += out %>
```

# Usage
1. Invoke `Templater: Create new note from template` Pallet command
2. Choose `OCR`
3. Choose an image file in your vault.  This may take a few seconds.

## [Video Demonstration](https://twitter.com/pmbauer/status/1392574345407500291)

[1]:https://github.com/SilentVoid13/Templater
[2]:https://obsidian.md
[3]:https://tesseract-ocr.github.io/
[4]:https://tesseract-ocr.github.io/tessdoc/Home.html#binaries

**Comment by ccchan234** [comment]
hi,
as the OP stated in the other thread,
this is NOT working in newer templater e.g. v.1.12 (see bottom)

could someone UPDATE it,
or CHANGE to use capture2text? thanks

PLEASE

https://forum.obsidian.md/t/basic-ocr-in-obsidian/18087

**Reply by pmbauer** [reply]
So all these scripts are made with volunteer time. Shouting your demands in all caps isn't motivating anyone to do you a favor.

**Reply by brandonzylstra** [reply]
> Add a User Function named `ocr`

@ccchan234  did you try adding a user function named `ocr` that uses `capture2text` instead of `tesseract`?

**Reply by ccchan234** [reply]
1. this thing is not use-able.

2. tell you more, gpt4 can help layman do programming, non use-able things like this should no longer exist

**Reply by brandonzylstra** [reply]
If you have so much confidence that it can be easily fixed with GPT4, why not fix it? 
It will accomplish much more than posting rude comments will.

**Reply by ccchan234** [reply]
> If you have so much confidence that it can be easily fixed with GPT4, why not fix it? It will accomplish much more than posting rude comments will.

to me, capture2text solved my issue.
i havenot complained being bothered by this useless script s post

**Reply by achtdev** [reply]
Thanks, for sharing your solution, not.

**Reply by alberto-tamez** [reply]
jeez man

**Comment by ccchan234** [comment]
foss doesn't make me silence, neither.

**Reply by pmbauer** [reply]
good luck with that
